libs/git/repository.go

```diff
@@ -23,8 +23,18 @@ type Repository struct {
 	// directory where we process .gitignore files.
 	real bool
 
-	// root is the absolute path to the repository root.
-	root vfs.Path
+	// checkoutDir is the path to the root of the repository checkout.
+	// This can be either the main repository checkout or a worktree.
+	checkoutDir vfs.Path
+
+	// gitDir is the equivalent of $GIT_DIR and points to the
+	// `.git` directory of a repository or a worktree.
```

Comment:
```suggestion
	// `.git` directory of a repository or the `.git` file for a worktree.
```

Reply:
It won't point to a file. If `.git` is a file (for worktrees), then it will contain a `gitdir: <xyz>` entry that points to the `$GIT_DIR`. This field contains this `<xyz>` reference.